README.md

```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
-
-## 3. SDK Reference
-### 3.1. Public Properties
-##### `String publisher`
-
-Mandatory. Sets the `publisher`  (can also be set via XML as `publisher`)
-
-##### `String mode`
-
-Mandatory. Sets the widget display `mode` (can also be set via XML as `mode`)
-
-##### `String placement`
-
-Mandatory. Sets the widget `placement` (can also be set via XML as `placement`)
-
-##### `String pageType`
-
-Mandatory. (Can also be set via XML as `page_type`)
-
-##### `String pageUrl`
-
-Mandatory. (Can also be set via XML as `url`)
-
-##### `String targetType`
-
-Optional. Default: `"mix"`. (can also be set via XML as `target_type`).
+### 2.5 Listening to ScrollToTop event
+A callback that notifies when the widget is on the top of the screen and is scrolled up (used for feed handling).
+#### 2.5.1 Registering a ScrollToTopListener
+This snippet shows how to register a scrollToTopListener
+     ```java
+      taboolaWidget.registerScrollToTopListener(new ScrollToTopListener() {
+              @Override
+              public void onTaboolaWidgetOnTop() {
+                  //Code..
+              }
+          });
+     ```
+
+#### 2.5.2 Unregistering a ScrollToTopListener
+This snippet shows how to unregister a ScrollToTopListener.
+     ```java
+      taboolaWidget.unregisterScrollToTopListener();
+     ```
+
+
+## 3. Example App
+This repository includes an example Android app which uses the Taboola SDK. 
+To use it:
+##### 3.1 Clone this repository
+1. Look for the "Clone or Download" button on this page top.
+2. Copy the url from the drop box.
+3. Clone to your local machine using your favourite Git client.
+
+##### 3.2 Open the project wih your IDE.
+1. Open the project as you would any other Android project.
+2. Taboola is optimized to working with Android Studio but other IDEs should work as well.
+
+##### 3.3 Example App As Troubleshooting Helper:
+In case you encounter some issues while integrating the SDK into your app, try to recreate the scenario within the example app. This might help to isolate the problems. For more help, you would be able to send the example app with your recreated issue to Taboola's support.
+
+## 4. SDK Reference - TaboolaWidget
+## `Notice: Dear reviewer, my definitions here might be incorrect/innacurate`
+##### `public TaboolaWidget setPublisher(String publisher)`
+**Mandatory**. Sets the `publisher` (The name of your application in Taboola Admin)
+
+##### `public TaboolaWidget setMode(String mode)`
+**Mandatory**. Sets the widget display `mode` (The name of your application in Taboola Admin)
+
+##### `public TaboolaWidget setPlacement(String placement)`
+**Mandatory**. Sets the widget `placement` (can also be set via XML as `placement`)
+
+##### `public TaboolaWidget setPageType(String pageType)`
+**Mandatory**. (Can also be set via XML as `page_type`)
```

Comment:
Given from Taboola's representatives, those are defining different type of pages as source and target and by those Taboola distinguish the recommendations required,

Reply:
This is correct for both source and target types.

t - text, v - video, s - search, p - photo, c - category, h - homepage, o - other, m - mix (a mix of item types, not a homogenous recommendation list)